py/DREAM/Output/SPIShardPositions.py

```diff
@@ -15,6 +17,70 @@ def __init__(self, name, data, grid, output, attr=list()):
         super().__init__(name=name, data=data, attr=attr, grid=grid, output=output)
         
 
+    def arrivalTime(self):
+        """
+        Estimate the time at which the pellet arrives to the plasma edge.
+        """
+        if 'eq' not in self.grid:
+            raise OutputException("Cannot plot poloidal trajectory when equilibrium data is not stored in output.")
+
+        R0 = self.grid.eq.R0
+        if np.isinf(R0): R0 = 1
+
+        ROverR0 = self.grid.eq.ROverR0_f*R0 - R0
+        Z = self.grid.eq.Z_f - self.grid.eq.Z0
+        ntheta = self.grid.eq.theta.size
+
+        vertices = [(ROverR0[i,-1], Z[i,-1]) for i in range(ntheta)]
+        p = path.Path(vertices)
+
+        xp = self.data[:,0::3,0]
+        yp = self.data[:,1::3,0]
+
+        # Check if the shards start within the plasma
+        if p.contains_point((xp[0,0], yp[0,0])):
+            return 0
```

Comment:
I assume that it is correct to just check if the first shard is in the plasma, and if it is not, no other shard will be either? Do they always start in the same position initially?

Reply:
All shards are launched from the same origin, so the idea here is that if one shard is inside the plasma, all are (because they are in the same place). It wouldn't work for a continuation simulation though, where the shards have already traveled a bit, so maybe we should generalize this as well?